soccer/planning/Path.hpp

```diff
@@ -164,12 +168,14 @@ class AngleFunctionPath : public Path {
      * @param[in]	shape The obstacles on the field
      * @param[out]  hitTime the approximate time when the path hits an obstacle.
      * If no obstacles are hit, behavior is undefined for the final value.
-     * @param[in] 	startTime The time on the path to start checking from
+     * @param[in] 	startTimeIntoPath The time on the path to start checking
+     *from
      * @return 		true if it hits an obstacle, otherwise false
      */
-    virtual bool hit(const Geometry2d::ShapeSet& obstacles, float& hitTime,
-                     float startTime) const override {
-        return path->hit(obstacles, hitTime, startTime);
+    virtual bool hit(const Geometry2d::ShapeSet& obstacles,
+                     RJ::Seconds startTimeIntoPath,
+                     RJ::Seconds* hitTime) const override {
```

Comment:
Well its an output variable so you want to give the option to not pass it in. Also some people (google which much of our style happens to be based off) rather encourage the use of pointers for output and only use of const references instead of mutable references.